packages/mantine/src/badge/Badge.tsx

```diff
@@ -28,6 +30,8 @@ export const Badge = forwardRef<
   // assertEmpty in this case is only used at typescript level, not runtime level
   assertEmpty(rest, false);
 
+  const { isOpened } = useContext(PopoverContext);
```

Comment:
@matthewlipski  we now introduce a knowledge from a "dumb" component (badge) about "what it should do when it's used in a specific way" (as part of a Popover).

Isn't this behavior (close a tooltip on click) more about the relation between the tooltip and the button (badge) instead of between the badge and the popover?

I think it can be cleaner if we have a tooltip that's more like the radix / shadcn one:

https://www.radix-ui.com/primitives/docs/components/tooltip : `https://www.radix-ui.com/primitives/docs/components/tooltip`